### PR TITLE
Empty Option added to DropDown and test cases added

### LINKED ISSUE #940: Add an empty option for DropDown Widget
**Is your feature request related to a problem? Please describe.**
Error correction for the [DropDown Widget](https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt) 

**Describe the solution you'd like**
I was planning to have something similar to #781 so that clicking on the drop-down choice would clear the answer.

**Describe alternatives you've considered**

**Additional context**
[ Choice questions should have clear options](https://github.com/google/android-fhir/pull/546#issuecomment-872723271
)

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #940

**Description**
1. Empty Option added for drop down widget 
2. New Test cases added for DropDown
3. Updated existing test cases

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by jingtang10** [comment]
can you rebase your pr once #995 is merged? thanks

**Comment by jingtang10** [comment]
> can you rebase your pr once #995 is merged? thanks

It's merged. @PallaviGanorkar plz rebase / merge. Also please remove the `@Ignore` that I added for the broken test. Thanks!

**Comment by PallaviGanorkar** [comment]
> can you rebase your pr once #995 is merged? thanks

done

**Comment by PallaviGanorkar** [comment]
> > can you rebase your pr once #995 is merged? thanks
> 
> It's merged. @PallaviGanorkar plz rebase / merge. Also please remove the `@Ignore` that I added for the broken test. Thanks!

Done

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        responseOptions()
+      ) {}
+    runOnUI { viewHolder.bind(questionnaireItemViewItem) }
+
+    onView(withId(R.id.auto_complete)).perform(showDropDown())
+    onView(withText("-"))
+      .inRoot(RootMatchers.isPlatformPopup())
+      .check(matches(isDisplayed()))
+      .perform(ViewActions.click())
+    Truth.assertThat(viewHolder.itemView.findViewById<TextView>(R.id.auto_complete).text.toString())
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:72]: import `Truth.assertThat`

```suggestion
   assertThat(viewHolder.itemView.findViewById<TextView>(R.id.auto_complete).text.toString())
```

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        responseOptions()
+      ) {}
+    runOnUI { viewHolder.bind(questionnaireItemViewItem) }
+
+    onView(withId(R.id.auto_complete)).perform(showDropDown())
+    onView(withText("Coding 3"))
+      .inRoot(RootMatchers.isPlatformPopup())
+      .check(matches(isDisplayed()))
+      .perform(ViewActions.click())
+    Truth.assertThat(viewHolder.itemView.findViewById<TextView>(R.id.auto_complete).text.toString())
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:90]: ditto

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      ) {}
+    runOnUI { viewHolder.bind(questionnaireItemViewItem) }
+
+    onView(withId(R.id.auto_complete)).perform(showDropDown())
+    onView(withText("-"))
+      .inRoot(RootMatchers.isPlatformPopup())
+      .check(matches(isDisplayed()))
+      .perform(ViewActions.click())
+    Truth.assertThat(viewHolder.itemView.findViewById<TextView>(R.id.auto_complete).text.toString())
+      .isEqualTo("-")
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:73]: can you please also test that the answer in the questionnaire response item is actually cleared?

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      ) {}
+    runOnUI { viewHolder.bind(questionnaireItemViewItem) }
+
+    onView(withId(R.id.auto_complete)).perform(showDropDown())
+    onView(withText("Coding 3"))
+      .inRoot(RootMatchers.isPlatformPopup())
+      .check(matches(isDisplayed()))
+      .perform(ViewActions.click())
+    Truth.assertThat(viewHolder.itemView.findViewById<TextView>(R.id.auto_complete).text.toString())
+      .isEqualTo("Coding 3")
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:91]: likewise, can you check coding 3 is actually set as answer?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt (hunk 1 above)
Comment (PallaviGanorkar) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:72]: Done

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt (hunk 2 above)
Comment (PallaviGanorkar) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:90]: Done

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt (hunk 4 above)
Comment (PallaviGanorkar) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:96]: Done

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt (hunk 3 above)
Comment (PallaviGanorkar) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactoryEspressoTest.kt:73]: Done

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**